lib/bridge/IrcBridge.js

```diff
@@ -754,6 +769,128 @@ IrcBridge.prototype.getBotClient = function(server) {
     });
 }
 
+
+var _solveTimeout = null;
+
+IrcBridge.prototype._updateVisibilityMap = function(isMode, key, value) {
+    let hasChanged = false;
+    if (isMode) {
+        if (typeof value !== 'boolean') {
+            throw new Error('+s state must be indicated with a boolean');
+        }
+        if (this._visibilityMap.channelModes[key] !== value) {
+            this._visibilityMap.channelModes[key] = value;
+            hasChanged = true;
+        }
+    }
+    else {
+        if (typeof value !== 'string' || (value !== "private" && value !== "public")) {
+            throw new Error('Room visibility must = "private" | "public"');
+        }
+
+        if (this._visibilityMap.roomVisibilities[key] !== value) {
+            this._visibilityMap.roomVisibilities[key] = value;
+            hasChanged = true;
+        }
+    }
+
+    if (hasChanged) {
+        clearTimeout(_solveTimeout);
+
+        _solveTimeout = setTimeout(() => {
+            this._solveVisibility()
+        }, 10000);
```

Comment:
We are doing cycle detection (by keeping a list of all channels checked in the stack, I'll push this in a bit). The reason this is used is so that if it is called again, the timeout won't fire. It resets the timer, meaning if we get 100 RPL_CHANMODEIS from 100 IRC clients joining, it will reset the timer 100 times, and then once 10 seconds of silence has elapsed, it will finally run.

If you think this behaviour is ok then I shall document it.